src/test/ui/consts/issue-79137-toogeneric.rs

```diff
@@ -0,0 +1,21 @@
+// The instrinsic call for variant count should exhaustively match on `tp_ty` and forbid
+// `ty::Projection`, `ty::Opaque`, `ty::Param`, `ty::Bound`, `ty::Placeholder` and `ty::Infer`
+// variant. This test checks that it will fail if it's too generic.
```

Comment:
Comments like this can easily get outdated, so it's often better to instead remain fairly implementation agnostic, for example by writing something like

```suggestion
// Test that `variant_count` only gets evaluated once the type is concrete enough.
```